spinnaker-jenkins-to-vmss/azuredeploy.json

```diff
@@ -0,0 +1,254 @@
+            }
+        },
+        "JenkinsUser": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of your Jenkins User account."
+            },
+            "defaultValue": "jenkins"
+        },
+        "JenkinsPassword": {
+            "type": "string",
+            "metadata": {
+                "description": "Password of your Jenkins User account."
+            }
+        },
+        "AptlyRepoName" : { 
+            "type": "string",
+            "metadata": {
+                "description": "Aptly Repo name"
+            }
+        },
+        "_artifactsLocation": {
+          "type": "string",
+          "metadata": {
+            "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+          },
+          "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/spinnaker-jenkins-to-vmss/"
+        },
+        "_artifactsLocationSasToken": {
+          "type": "securestring",
+          "metadata": {
+            "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+          },
+          "defaultValue": ""
+        }
+    },
+  "variables": {
+    "scenarioPrefix": "spinnaker",
+    "storageAccountName": "[concat('vhdstorage', uniqueString(resourceGroup().id))]",
+    "front50StorageAccountName": "[concat('front50', uniqueString(resourceGroup().id))]",
+    "storageAccountContainerName": "vhds",
+    "packerStorageAccountName": "[concat('packer', uniqueString(resourceGroup().id))]",
+    "spinnakerTemplateUrl": "[concat(parameters('_artifactsLocation'), 'nested/', 'spinnakerDeploy.json', parameters('_artifactsLocationSasToken'))]",
+    "storageTemplateUrl": "[concat(parameters('_artifactsLocation'), 'nested/', 'storageDeploy.json', parameters('_artifactsLocationSasToken'))]",
+    "vnetTemplateUrl": "[concat(parameters('_artifactsLocation'), 'nested/', 'vnetDeploy.json', parameters('_artifactsLocationSasToken'))]",
+    "jenkinsTemplateUrl": "[concat(parameters('_artifactsLocation'), 'nested/', 'jenkinsDeploy.json', parameters('_artifactsLocationSasToken'))]",
+    "vnetName": "[concat(variables('scenarioPrefix'),'Vnet')]",
+    "subnet0Name": "[concat(variables('scenarioPrefix'),'Subnet0')]",
+    "subnet1Name": "[concat(variables('scenarioPrefix'),'Subnet1')]",
+    "subnet2Name": "[concat(variables('scenarioPrefix'),'Subnet2')]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))]",
+    "subnet0Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet0Name'))]",
+    "jenkinsFQDN": "[concat(parameters('jenkinsDnsPrefix'), '.', resourceGroup().location, '.cloudapp.azure.com' )]"
```

Comment:
Moving to a single template simplifies the issue, I am now using: reference('jenkinsPublicIP').dnsSettings.fqdn